src/Core/EnvironmentSettings.cs

```diff
@@ -25,14 +25,36 @@ namespace SonarLint.VisualStudio.Core
     public class EnvironmentSettings : IEnvironmentSettings
     {
         internal const string TreatBlockerAsErrorEnvVar = "SONAR_INTERNAL_TREAT_BLOCKER_AS_ERROR";
+        internal const string CFamilyAnalysisTimeoutEnvVar = "SONAR_INTERNAL_CFAMILY_ANALYSIS_TIMEOUT_MS";
+        public const string SonarLintDownloadUrlEnvVar = "SONARLINT_DAEMON_DOWNLOAD_URL";
 
-        public  bool TreatBlockerSeverityAsError()
+        public bool TreatBlockerSeverityAsError()
         {
             if (bool.TryParse(Environment.GetEnvironmentVariable(TreatBlockerAsErrorEnvVar), out var result))
             {
                 return result;
             }
             return false;
         }
+
+        public int CFamilyAnalysisTimeoutInMs()
+        {
+            var setting = Environment.GetEnvironmentVariable(CFamilyAnalysisTimeoutEnvVar);
+
+            if (int.TryParse(setting, System.Globalization.NumberStyles.Integer, System.Globalization.NumberFormatInfo.InvariantInfo, out int userSuppliedTimeout)
+                && userSuppliedTimeout > 0)
+            {
+                return userSuppliedTimeout;
+            }
+
+            return 0;
+        }
+
+        public string SonarLintDaemonDownloadUrl()
+        {
+            // The URL validation and logging is being done by the daemon installer, so
+            // this is just a passthrough
+            return Environment.GetEnvironmentVariable(SonarLintDownloadUrlEnvVar);
```

Comment:
I wanted to move the URL validation here, but unfortunately that was complicated by the fact that the daemon installer is logging detailed messages about invalid URLs.
In the end I just left this method as a passthrough for now, so at least we have a level of abstraction that makes the daemon installer tests simpler.